plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1660,14 +1652,14 @@ private void commitShared()
             this.transaction = transaction;
         }
 
-        private void prepareDropTable(SchemaTableName schemaTableName)
+        private void prepareDropTable(SchemaTableName schemaTableName, boolean deleteData)
         {
             metastoreDeleteOperations.add(new IrreversibleMetastoreOperation(
                     format("drop table %s", schemaTableName),
                     () -> {
                         Optional<Table> droppedTable = delegate.getTable(schemaTableName.getSchemaName(), schemaTableName.getTableName());
                         try {
-                            delegate.dropTable(schemaTableName.getSchemaName(), schemaTableName.getTableName(), true);
+                            delegate.dropTable(schemaTableName.getSchemaName(), schemaTableName.getTableName(), deleteData);
```

Comment:
Just in case somebody else is looking for it, `io.trino.plugin.hive.BaseHiveConnectorTest#testCreateExternalTable(java.lang.String, java.lang.String, java.lang.String, java.util.List<java.lang.String>)` verifies that the location still exists while trying to drop an `EXTERNAL` table.